homeassistant/components/freebox/sensor.py

```diff
@@ -98,7 +99,19 @@ async def async_setup_entry(
         for description in DISK_PARTITION_SENSORS
     )
 
-    async_add_entities(entities, True)
+    for node in router.home_devices.values():
+        battery_node: dict[str, Any] = next(
+            filter(
```

Comment:
I don't really see a reason to use a filter and lambda and all that instead of a for loop, this makes it look very complex. But opinions differ, so is there any reason why this was added?

Reply:
In `router.home_devices` there are all devices supported by the Freebox Home API, not only device batteries, but also cameras (wich are supported by HA), covers, PIRs, locks, remots ... (which support should arrive in HA).

Filter is wanted here.

Reply:
But what's the difference between a filter and a for loop and an if statement?

Reply:
Oh yeah, for-if loop looks prettier and future proof (when adding new sensor) 😌